src/main/java/com/emc/object/util/RestUtil.java

```diff
@@ -206,6 +211,23 @@ public static Date headerParse(String dateString) {
         }
     }
 
+    public static Date headerParse(String dateString, String headerKey) {
+        if (dateString == null) return null;
+        try {
+            if(headerKey.equals(S3Constants.AMZ_DATE)) {
+                // convert date
+                SimpleDateFormat sdf = new SimpleDateFormat(AMZ_DATE_FORMAT);
```

Comment:
this is an expensive operation - we should use a `ThreadLocal` reference, like the other dateformats at the top of this class - in fact, it may be better to use a static initializer, or initialize in `ThreadLocal.withInitial()`, and get rid of the `getHeaderFormat()` static method

Reply:
Changed.